app/src/main/java/ca/klapstein/baudit/data/Problem.java

```diff
@@ -41,6 +43,17 @@ static public boolean isValidProblemTitle(String title) {
     public RecordTreeSet getRecordTreeSet() {
         return recordTreeSet;
     }
+    
+    public Problem(@NonNull String title, String description) throws IllegalArgumentException{
+        this.setTitle(title);
+        this.setDescription(description);
+        this.date = new Date();
+    }
+
+     @Override
+    public int compareTo(@NonNull Problem problem) {
+      return this.getTitle().compareTo(problem.getTitle()); //Order by date
```

Comment:
This does not order by Date, it orders by Title.

I would be best if we sort problems by Date. And if two problem's dates are the same we fallback onto sorting by Title.